src/signer.rs

```diff
@@ -6,10 +6,22 @@ use rustls::{
     sign::{Signer, SigningKey},
     Error, OtherError, SignatureAlgorithm, SignatureScheme,
 };
-use sha2::digest::Digest;
+extern crate bcrypt;
+extern crate winapi;
+extern crate windows_sys;
+
+///use sha2::digest::Digest;
 
 use crate::key::{AlgorithmGroup, NCryptKey, SignaturePadding};
 
+use winapi::shared::bcrypt::BCryptHash;
+use windows_sys::Win32::Security::Cryptography::BCRYPT_ALG_HANDLE;
+
+use std::ptr::null_mut;
+const BCRYPT_SHA256_ALG_HANDLE: BCRYPT_ALG_HANDLE = 0x00000041 as BCRYPT_ALG_HANDLE;
+const BCRYPT_SHA384_ALG_HANDLE: BCRYPT_ALG_HANDLE = 0x00000051 as BCRYPT_ALG_HANDLE;
+const BCRYPT_SHA512_ALG_HANDLE: BCRYPT_ALG_HANDLE = 0x00000061 as BCRYPT_ALG_HANDLE;
```

Comment:
You definitely shouldn't be defining these.

Should be available in the windows_sys crate.

Reply:
done